src/landing_page/src/pages/landing/landing.js

```diff
@@ -49,54 +48,66 @@ function Landing() {
   useEffect(() => {
     // Fetch the DMP ID metadata from the DMPHub
     let api = new DmpApi();
-
-    fetch(api.getUrl(), api.getOptions())
-      .then((resp) => {
-        api.handleResponse(resp);
-        return resp.json();
-      })
-      .then((data) => {
-        // console.log(data.items[0].dmp);
+    const controller = new AbortController();
+    const fetchData = async () => {
+      try{
+        const response = await fetch(api.getUrl(),api.getOptions(),{signal:controller.signal});
+        api.handleResponse(response);
+        const data = await response.json();
         if (Array.isArray(data?.items) && data?.items[0] !== null) {
           let dmp = data.items[0].dmp;
-          //console.log(dmp);
 
           setFormData({
             json_url: api.getUrl(),
-            title: getValue(dmp, "title", ""),
-            description: getValue(dmp, "description", ""),
-            dmp_id: getValue(dmp, "dmp_id.identifier", ""),
-            privacy: getValue(dmp, "dmproadmap_privacy", "private"),
-            created: getValue(dmp, "created", ""),
-            modified: getValue(dmp, "modified", ""),
-            ethical_issues_exist: getValue(dmp, "ethical_issues_exits", "unknown"),
-            ethical_issues_report: getValue(dmp, "ethical_issues_report", ""),
-
-            funder_name: getValue(dmp, "project.0.funding.0.name", ""),
-            funder_id: getValue(dmp, "project.0.funding.0.funder_id.identifier", ""),
-            award_id: getValue(dmp, "project.0.funding.0.grant_id.identifier", ""),
-            opportunity_number: getValue(dmp, "project.0.funding.0.dmproadmap_funding_opportunity_id.identifier", ""),
-
-            project_title: getValue(dmp, "project.0.title", ""),
-            project_abstract: getValue(dmp, "project.0.description", ""),
-            project_start: getValue(dmp, "project.0.start", ""),
-            project_end: getValue(dmp, "project.0.end", ""),
-
-            contact: getValue(dmp, "contact", {}),
-            contributors: getValue(dmp, "contributor", []),
-            datasets: getValue(dmp, "dataset", []),
-            related_identifiers: getValue(dmp, "dmproadmap_related_identifiers", []),
-            versions: getValue(dmp, "dmphub_versions", []),
+            title: dmp.title || "",
+            description: dmp.description || "",
```

Comment:
Ah I see, so this removes all the unnecessary overhead of the getValue() function. I probably just blindly copy-pasted that from the work the consultants did, but it makes sense on their page because the user is actually interacting with the data

Reply:
Yea, it wasn't needed here and I was worried that the recursive functionality was adding unnecessary time and space complexity.